### PR TITLE
Add jupyter functions and examples

### PR DESCRIPTION
**Pull request by psavery**
This adds functions and examples for running trame in either a jupyter server proxy process or
within the jupyter event loop.

Depends on: kitware/trame#54

### CODE REVIEW

File: {{cookiecutter.package_name}}/{{cookiecutter.import_name}}/app/jupyter.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from {{cookiecutter.import_name}}.app import controller, ui
+
+
+def run_in_loop(**kwargs):
+    """Run and display the trame application in jupyter's event loop
+
+    The kwargs are forwarded to IPython.display.IFrame()
+    """
+    controller.on_start()
+    ui.layout.start(exec_mode="task", server=True)
```
Comment (psavery) [inline {{cookiecutter.package_name}}/{{cookiecutter.import_name}}/app/jupyter.py:15]: FYI: one problem with having these two lines of code here is that we only want to run these two lines of code once. So if you re-run the `run_in_loop()` function again, we get some exceptions.

Comment (jourdain) [reply {{cookiecutter.package_name}}/{{cookiecutter.import_name}}/app/jupyter.py:20]: you can probably guard them with a global flag within that module

**Review by jourdain — APPROVED** [review]
LGTM but I would rather have the width=100% the default and just have the user provide a height. That seems a more reasonable scenario.

File: {{cookiecutter.package_name}}/{{cookiecutter.import_name}}/app/jupyter.py (hunk 1 above)
Comment (psavery) [reply {{cookiecutter.package_name}}/{{cookiecutter.import_name}}/app/jupyter.py]: Done.

**Comment by psavery** [comment]
> LGTM but I would rather have the width=100% the default and just have the user provide a height. That seems a more reasonable scenario.

Done.

**Pull request merged**